Libs/MRML/Logic/vtkMRMLAbstractVolumeResampler.h

```diff
@@ -0,0 +1,93 @@
+/*==============================================================================
+
+  Program: 3D Slicer
+
+  Copyright(c) Kitware Inc.
+
+  See COPYRIGHT.txt
+  or http://www.slicer.org/copyright/copyright.txt for details.
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+==============================================================================*/
+
+#ifndef __vtkMRMLAbstractVolumeResampler_h
+#define __vtkMRMLAbstractVolumeResampler_h
+
+// MRMLLogic includes
+#include "vtkMRMLLogicExport.h"
+
+// MRML includes
+class vtkMRMLTransformNode;
+class vtkMRMLVectorVolumeNode;
+class vtkMRMLVolumeNode;
+
+// VTK includes
+#include <vtkObject.h>
+
+/// \brief Base class for volume resampler.
+///
+/// Sub-classes must implement actual resampling.
+class VTK_MRML_LOGIC_EXPORT vtkMRMLAbstractVolumeResampler : public vtkObject
+{
+public:
+  vtkTypeMacro(vtkMRMLAbstractVolumeResampler, vtkObject);
+  void PrintSelf(ostream& os, vtkIndent indent) override;
+
+  enum
+  {
+    InterpolationTypeUndefined,
```

Comment:
@lassoan  To follow up on this morning discussion, I added explicit enums for interpolation type and window function.